image/image.go

```diff
@@ -380,17 +407,22 @@ func setupSeed(tsto *ToolingStore, model *asserts.Model, opts *Options) error {
 	}
 
 	// unpacking the gadget for core models
-	if err := unpackGadget(gadgetFname, opts.GadgetUnpackDir); err != nil {
+	if err := unpackGadget(gadgetFname, gadgetUnpackDir); err != nil {
 		return err
 	}
 
-	if err := boot.MakeBootable(model, opts.RootDir, bootWith); err != nil {
+	// TODO|XXX: change MakeBootable/pass right info Core 20 case
+	// (setting up recovery, not run bootenv)
+	if err := boot.MakeBootable(model, bootRootDir, bootWith); err != nil {
 		return err
 	}
 
-	// and the cloud-init things
-	if err := installCloudConfig(opts.RootDir, opts.GadgetUnpackDir); err != nil {
-		return err
+	// cloud-init config (done at install for Core 20)
+	if !core20 {
+		// and the cloud-init things
+		if err := installCloudConfig(rootDir, gadgetUnpackDir); err != nil {
+			return err
+		}
 	}
 
 	return nil
```

Comment:
reading this function start to end makes me feel like it needs to be split up... not for this PR, but a 250 line long function is not a fun thing to debug and track :confused: